### DISCUSSION TITLE
Single public email

### DISCUSSION THREAD

**Discussion by FMorschel** [community/community#31604] [Other Feature Feedback, Questions, & Ideas]
I would like to have my emails linked to my account to be private for security and privacy. But I would love the option of enabling one single email to be public so that people visiting my profile can contact me. I've solved it for now by posting it on my bio, but that's not an optimal solution at all.

**Comment by FMorschel** [comment]
Just so that I'm clear, I wanted to set my account emails as private to disable any commits from my personal emails. If I unmark that my emails are private, that option goes away.

**Reply by Habeel06** [reply]
You can just use your public email on your github for commits.

**Reply by Jackenmen** [reply]
Of course, you can, the problem is that GitHub will not prevent you from accidentally exposing your email, see my other comment: https://github.com/orgs/community/discussions/31604#discussioncomment-4118871

**Comment by Zwyx** [comment]
I provide my support for this feature request.

- I'd like to keep my email addresses private, so I can tick the box **Block command line pushes that expose my email**,
- except one address, that would be shown on my public profile.

**Reply by Chesney97** [reply]
Thia dosant really the queations guy

**Comment by Zwyx** [comment]
@FMorschel, note that GitHub doesn't show email addresses on profile pages if the user browsing isn't authenticated. A (small) protection against spamming, I believe. By putting your email address in your bio, you lose this protection.

**Comment by alekdev0** [comment]
Hello @FMorschel 
You can add  your private email like following:
1. Please login your github account.
2. Please go to your profile (click "your profile" under your avator )
3. And then please add your public email address in Public Profile >> Public email

**Reply by FMorschel** [reply]
I know this is how. The thing is that I still want **Block command line pushes that expose my email** to be on. And only one of my emails is actually a contact one. That one I'm willing to leave on my profile. But Since that option is activated, I cannot do as you say because it blocks me from "exposing" my email.

**Reply by alekdev0** [reply]
Please check my profile, So do you want to update email setting as mine?

**Reply by Jackenmen** [reply]
@WHdev711 The problem is that nothing prevents you from accidentally exposing your other emails in your setup. For example:
- it still shows up in the web UI when doing git actions:
![image](https://user-images.githubusercontent.com/6032823/201381511-0b44adea-92d4-4c22-882c-24d8ccaa989b.png)
- you can still accidentally use it locally and push commits with it
- you can reply to notification emails with it and expose your email this way
- sort of related, your public emails NEEDS TO be your primary email (and therefore can be used for password resets which is a potential security problem) if you want to make sure that GitHub won't ever implicitly use your private email for commits, see #32043

**Reply by FMorschel** [reply]
Exactly what I am trying to explain, thank you @jack1142!

**Reply by alekdev0** [reply]
Good

**Comment by Chesney97** [comment]
Jusz gi my page only thing is my number may get pruvate for that lol

**Reply by Chesney97** [reply]
Go to my page info their

**Comment by jfullstackdev** [comment]
I'm wondering if this is the one you are referring to, the `noreply` email

![image](https://user-images.githubusercontent.com/47092464/201473134-6806fad8-3643-4026-954b-9a435d4aa2c9.png)

https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-email-preferences/setting-your-commit-email-address

**Reply by FMorschel** [reply]
I do use this email as a commit email, but I wanted to expose my contact email on my profile and not sync it with my GitHub login information. And disabling my commits with the emails synchronized with my account.

**Reply by jfullstackdev** [reply]
I see you are using the default README, simply put your public email for contact at that and you're done

**Comment by Rolandjosh** [comment]
> I would like to have my emails linked to my account to be private for security and privacy. But I would love the option of enabling one single email to be public so that people visiting my profile can contact me. I've solved it for now by posting it on my bio, but that's not an optimal solution at all.

He
